Directory.Build.props

```diff
@@ -202,10 +202,28 @@
     <RootRepo>known-good</RootRepo>
     <RootRepoTests>known-good-tests</RootRepoTests>
   </PropertyGroup>
-
-  <ItemGroup>
-    <SourceBuiltPackage Include="$(SourceBuiltPackagesPath)/*.nupkg" />
-  </ItemGroup>
+  
+  <!--
+    Semaphore paths for incremental builds
+  -->
+  <PropertyGroup>
+    <CompletedSemaphorePath>$(BaseIntermediatePath)semaphores/</CompletedSemaphorePath>
+    <AcquireDarcSuccessSemaphore>$(CompletedSemaphorePath)AcquireDarc.complete</AcquireDarcSuccessSemaphore>
+    <UnpackTarballsSuccessSemaphore>$(CompletedSemaphorePath)UnpackTarballs.complete</UnpackTarballsSuccessSemaphore>
+    <BuildTasksSuccessSemaphore>$(CompletedSemaphorePath)BuildTasks.complete</BuildTasksSuccessSemaphore>
+    <BuildXPlatTasksSuccessSemaphore>$(CompletedSemaphorePath)BuildXPlatTasks.complete</BuildXPlatTasksSuccessSemaphore>
+    <BuildReferenceAssembliesSuccessSemaphore>$(CompletedSemaphorePath)BuildReferenceAssemblies.complete</BuildReferenceAssembliesSuccessSemaphore>
+    <ExtractToolPackageSuccessSemaphore>$(CompletedSemaphorePath)ExtractToolPackage.complete</ExtractToolPackageSuccessSemaphore>
+    <WriteDynamicPropsToStaticPropsFilesSuccessSemaphore>$(CompletedSemaphorePath)WriteDynamicPropsToStaticPropsFiles.complete</WriteDynamicPropsToStaticPropsFilesSuccessSemaphore>
+    <PoisonPrebuiltPackagesSuccessSemaphore>$(CompletedSemaphorePath)PoisonPrebuiltPackages.complete</PoisonPrebuiltPackagesSuccessSemaphore>
+    <FetchSourcesSuccessSemaphore>$(CompletedSemaphorePath)FetchSources.complete</FetchSourcesSuccessSemaphore>
+    <WriteSourceLinkMetadataSuccessSemaphore>$(CompletedSemaphorePath)WriteSourceLinkMetadata.complete</WriteSourceLinkMetadataSuccessSemaphore>
+    <CopyAndDisassembleReferenceOnlyPackagesSuccessSemaphore>$(CompletedSemaphorePath)CopyAndDisassembleReferenceOnlyPackages.complete</CopyAndDisassembleReferenceOnlyPackagesSuccessSemaphore>
+    <DownloadSourceBuildReferencePackagesSuccessSemaphore>$(CompletedSemaphorePath)DownloadSourceBuildReferencePackages.complete</DownloadSourceBuildReferencePackagesSuccessSemaphore>
+    <DownloadSourceBuiltArtifactsSuccessSemaphore>$(CompletedSemaphorePath)DownloadSourceBuiltArtifacts.complete</DownloadSourceBuiltArtifactsSuccessSemaphore>
+    <ReportPoisonUsageSuccessSemaphore>$(CompletedSemaphorePath)ReportPoisonUsage.complete</ReportPoisonUsageSuccessSemaphore>
+    <GeneratePrebuiltBurndownDataSuccessSemaphore>$(CompletedSemaphorePath)GeneratePrebuiltBurndownData.complete</GeneratePrebuiltBurndownDataSuccessSemaphore>
+  </PropertyGroup>
```

Comment:
I'm not convinced about the maintainability of having props for these... it seems like this is basically:

```xml
<PropertyGroup>
  <GeneratePrebuiltBurndownDataSuccessSemaphore>$(CompletedSemaphorePath)GeneratePrebuiltBurndownData.complete</GeneratePrebuiltBurndownDataSuccessSemaphore>
</PropertyGroup>
<Target Name="GeneratePrebuiltBurndownData"
        Inputs="$(MSBuildProjectFullPath)" 
        Outputs="$(GeneratePrebuiltBurndownDataSuccessSemaphore)">
  ...
  <Touch Files="$(GeneratePrebuiltBurndownDataSuccessSemaphore)" AlwaysCreate="true" />
</Target>
```

vs.

```xml
<Target Name="GeneratePrebuiltBurndownData"
        Inputs="$(MSBuildProjectFullPath)" 
        Outputs="$(CompletedSemaphorePath)GeneratePrebuiltBurndownData.complete">
  ...
  <Touch Files="$(CompletedSemaphorePath)GeneratePrebuiltBurndownData.complete" AlwaysCreate="true" />
</Target>
```

I prefer the second by quite a bit:
* I can double click the target name and paste it into the paths.
  * It's hard to rename a target with the non-delimited names currently.
* The info isn't spread out across files.
  * In theory, you can put a property group just above the target like I did above, to keep them together but still use the current approach. This would be non-standard MSBuild though.

Are there benefits to the current way I'm not seeing?

Reply:
My thinking for adding them all in props was to make sure that there weren't duplicate names of semaphore files, because this could cause hard to diagnose problems.  Thinking more about it, I like your suggestion and paring it down to less code to implement.  grep can ensure that duplicates don't crop up.

Reply:
That's a good point, I think that the extra maintenance is probably more risky than that though, and it's somewhat mitigated by MSBuild target names typically having to be unique anyway (or worse things will happen).

Could consider writing a build task that parses our MSBuild files and looks for unique and matching values to go even beyond. (I think the MSBuild libraries have a parser we could use. Seemed nice when I saw it but completely forget the context.)